### DISCUSSION TITLE
Support for GHA cache?

### DISCUSSION THREAD

**Discussion by ymarcus93** [testcontainers/testcontainers-go#2119] [Q&A]
Is there a way to configure `testcontainers-go` to use the GitHub Cache API to cache builds across workflow runs? Something like this: https://docs.docker.com/build/ci/github-actions/cache/#cache-backend-api

**Comment by mdelapenya** [comment]
Hey @ymarcus93 I'm afraid that tescontainers-go's responsibilities are not to cache things at the CI level, but I think you could use this GH action instead: https://github.com/marketplace/actions/docker-cache?

Would that work for you?

**Reply by ymarcus93** [reply]
Hey @mdelapenya! Thanks for the recommendation. Unfortunately, I don't think the linked action solves my usecase entirely. It seems to only cache built Docker images, but what I really want is to cache the intermediate layers that comprise the image.

> Note that this action does not perform Docker layer caching. The official Docker build push action performs Docker layer caching for built images but does not cache pulled images.

I can give more context if that helps. Essentially, I'm writing some integration tests with a `Dockerfile` that modifies a base image from the Docker registry to setup some dependencies needed for the test, and then the final steps are building the binary/sut hosted in the same repo.

So even if I use this action to cache images, `testcontainers` will still re-build the entire image and not reuse any layers from the cache because it only sees previously built images, not the layers.

I think there might be a way to get it to work if I separate the modifications of the base image imported from Docker hub registry into a separate `Dockerfile` and have that be cached, but I'm not sure. Ideally, I really just need the layers to be cached.
